olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/VersionedLayerClient.h

```diff
@@ -375,6 +375,24 @@ class DATASERVICE_READ_API VersionedLayerClient final {
    */
   bool RemoveFromCache(const geo::TileKey& tile);
 
+  /**
+   * @brief Checks if partition is cached.
+   *
+   * @param partition_id The partition ID that we check.
+   *
+   * @return True if partition data is cached; false otherwise.
+   */
+  bool IsCached(const std::string& partition_id) const;
+
+  /**
+   * @brief Checks if tile is cached.
+   *
+   * @param tile The tile key that we check.
+   *
+   * @return True if tile data is cached; false otherwise.
+   */
```

Comment:
```suggestion
  /**
   * @brief Checks whether the tile is cached.
   *
   * @param tile The tile key.
   *
   * @return True if the tile data is cached; false otherwise.
   */
```